File: ps/zziri/leetcode/4. Median of Two Sorted Arrays.py

class Solution:
    def merge(self, arr1, arr2):
        ret = list()
        i, j = 0, 0
        while i < len(arr1) and j < len(arr2):
            if arr1[i] < arr2[j]:
                ret.append(arr1[i])
                i += 1
            else:
                ret.append(arr2[j])
                j += 1
        if i >= len(arr1):
            ret.extend(arr2[j:])
        if j >= len(arr2):
            ret.extend(arr1[i:])
        
        return ret
    
    def get_median(self, arr):
        print(arr)
        mid = int(len(arr)/2)
        if len(arr) % 2 == 0:
            return (arr[mid] + arr[mid-1])/2
        else:
            return arr[mid]
            
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        merged = self.merge(nums1, nums2)
        return self.get_median(merged)

File: ps/rewyear/leetcode/4.Median_of_Two_Sorted_Arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int len1 = nums1.length;
        int len2 = nums2.length;
        int mid = (len1 + len2) / 2;
        
        int[] merged = new int[len1 + len2];
        
        System.arraycopy(nums1,0,merged,0,len1);
        System.arraycopy(nums2,0,merged,len1,len2);
        
        Arrays.sort(merged);
        
        if(merged.length % 2 == 0)
            return (merged[mid] + merged[mid-1])/2.0;
        else
            return merged[mid];
        
    }
}
